akka-docs/src/test/java/jdocs/stream/operators/SourceOrFlow.java

```diff
@@ -354,28 +354,53 @@ private Histogram(long low, long high) {
     // Immutable start value
     public static Histogram INSTANCE = new Histogram(0L, 0L);
 
+    // #foldAsync
+
     public Histogram add(int number) {
       if (number < 100) {
         return new Histogram(low + 1L, high);
       } else {
         return new Histogram(low, high + 1L);
       }
     }
+    // #fold
+
+    // #foldAsync
+    public CompletionStage<Histogram> addAsync(Integer n) {
+      if (n < 100) {
+        return CompletableFuture.supplyAsync(() -> new Histogram(low + 1L, high));
+      } else {
+        return CompletableFuture.supplyAsync(() -> new Histogram(low, high + 1L));
+      }
+    }
+    // #fold
```

Comment:
I would have wished for a less synthetic async operation, but I know coming up with that is tricky.

One idea would be to do a (made up) actor ask for the aggregate instead of re-using the sync fold datastructure.

Reply:
Hi @johanandren, 
Do we have any such example which I can refer to address this comment?

Reply:
I don't think there is such an example anywhere yet, the [Source.unfoldResourceAsync](https://doc.akka.io/docs/akka/current/stream/operators/Source/unfoldResourceAsync.html#source-unfoldresourceasync) has a imaginary async database call to show off intended usage, maybe that can be an inspiration.